### PR TITLE
ddl: support online create multi-valued index

### PR DESCRIPTION
**Pull request by xiongjiwei**
Signed-off-by: xiongjiwei <xiongjiwei1996@outlook.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/40337

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/mv_index_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl_test
+
+import (
+	"fmt"
+	"github.com/pingcap/tidb/ddl"
+	"github.com/pingcap/tidb/parser/model"
```
Comment (hawkingrei) [inline ddl/mv_index_test.go:20]: need to goimports.

Comment (xiongjiwei) [reply ddl/mv_index_test.go]: done

**Comment by xiongjiwei** [comment]
@YangKeao @tangenta @Defined2014
please help review

File: table/tables/index.go [hunk 2]
```diff
@@ -402,6 +402,46 @@ func (c *index) Delete(sc *stmtctx.StatementContext, txn kv.Transaction, indexed
 	return nil
 }
 
+func (c *index) GenIndexKVIter(sc *stmtctx.StatementContext, indexedValue []types.Datum, h kv.Handle, handleRestoreData []types.Datum, untouched bool) table.IndexIter {
```
Comment (Defined2014) [inline table/tables/index.go:405]: Do we need `untouched` param? Seems it not be used here.

Comment (xiongjiwei) [reply table/tables/index.go:405]: Ah, I will remove it

**Review by CbcWestwolf — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

File: ddl/index.go [hunk 3]
```diff
@@ -1548,12 +1562,7 @@ func (w *addIndexWorker) batchCheckUniqueKey(txn kv.Transaction, idxRecords []*i
 		} else if w.distinctCheckFlags[i] {
 			// The keys in w.batchCheckKeys also maybe duplicate,
 			// so we need to backfill the not found key into `batchVals` map.
```
Comment (xhebox) [inline ddl/index.go:1564]: Maybe move this comment to `GenIndexValue`

**Review by xhebox — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**